### DISCUSSION TITLE
"Error: secret "infinispan-secret" not found" error on K8s

### DISCUSSION THREAD

**Discussion by pai911** [apache/camel-karavan#1057] [Q&A]
I'm seeing the error after trying to install Karavan on my company's k8s cluster

It seems someone has talked about how to fix the issue (
https://github.com/apache/camel-karavan/discussions/923), but I still don't get it how to do it?

I've added these params...

--install-infinispan 
--infinispan-image quay.io/infinispan/server:14.0.21.Final 
--infinispan-username user 
--infinispan-password pass

However, I still saw `Warning  Failed     5s (x3 over 17s)  kubelet            Error: secret "infinispan-secret" not found.`

What exactly is this infinispan used for in Karavan? Is it really required?

Pls shed some lights on this

**Comment by mgubaidullin** [comment]
`infinispan-secret` installed by Infinispan to store Infinispan username and password.
Karavan uses Infinispan as a cache for all user data before user commit and push it to git. And also for all container/deployment statuses.

**Reply by pai911** [reply]
so can I say it's not mandatory if I don't want to push it to Git?

Why can't I ignore it in my installation? 

Is there a document on how to set these options correctly?

**Comment by mgubaidullin** [comment]
Karavan works the way I described it.
It uses Infinspan to cache all user data.
It uses git repository as a storage.
Build container pulls code from git to build and deploy integration microcontainer.

By default karavan-installer installs infinispan with default username and password using the following YAML https://github.com/apache/camel-karavan/blob/e5f5989a227c9f0630d46fba3ef3b83b25844db0/karavan-web/karavan-installer/src/main/resources/infinispan.yaml#L18C13-L18C13 

If you want to set custom username and password use infinispan parameters described here https://github.com/apache/camel-karavan/blob/main/docs/WEB_KUBERNETES.md#install-parameters

**Reply by pai911** [reply]
Thank you for the reply!

I'm okay with using the default, but the problem is if I ignore all the related params ...

```
--install-infinispan
--infinispan-image
--infinispan-username
--infinispan-password
```

I see the following error and the pod can't be created.

```
"Warning  Failed     5s (x3 over 17s)  kubelet            Error: secret "infinispan-secret" not found."
```

Do you have idea on what's happening? 

I followed this guide (https://github.com/apache/camel-karavan/blob/main/docs/WEB_KUBERNETES.md). Did I miss any key step? Thank you!
